core/src/components/datetime/datetime-util.ts

```diff
@@ -284,14 +285,23 @@ export const getLocalDateTime = (dateString: any = ''): Date => {
       date.getMilliseconds()
     )
   );
+
+  if (timeZone && timeZone.length > 0) {
+    const utcDateTime = new Date(localDateTime.toLocaleString('en-US', { timeZone: 'utc' }));
+    const tzDateTime = new Date(localDateTime.toLocaleString('en-US', { timeZone }));
+    const offset = utcDateTime.getTime() - tzDateTime.getTime();
```

Comment:
Another thought: Let's try moving these three lines to a util function in `datetime-util.ts`. I see we are using this code in two different places, so moving it to the util would be good.

Something like:

```typescript
const getTimezoneOffset = (date: Date, timezone: string) => {
  const utcDateTime = new Date(localDateTime.toLocaleString('en-US', { timeZone: 'utc' }));
  const tzDateTime = new Date(localDateTime.toLocaleString('en-US', { timeZone }));
  return utcDateTime.getTime() - tzDateTime.getTime();
}
```